### PR TITLE
deps: update ANTLR and cel-cpp

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Commit Message: update ANTLR to 4.10.1 and CEL to latest commit
Risk Level: low
Testing: regression
Docs Changes: none
Release Notes: none

### CODE REVIEW

File: bazel/repositories.bzl [hunk 1]
```diff
@@ -495,26 +495,6 @@ def _com_github_facebook_zstd():
...
-cc_library(
-    name = "cpp",
-    srcs = glob(["runtime/Cpp/runtime/src/**/*.cpp"]),
-    hdrs = glob(["runtime/Cpp/runtime/src/**/*.h"]),
-    includes = ["runtime/Cpp/runtime/src"],
-)
-""",
-        patch_args = ["-p1"],
-        # Patches ASAN violation of initialization fiasco
-        patches = ["@envoy//bazel:antlr.patch"],
```
Comment (keith) [inline bazel/repositories.bzl:516]: can you delete this patch as well?

Comment (kyessenov) [reply bazel/repositories.bzl:516]: done, thanks.

**Review by keith — DISMISSED** [review]

**Comment by kyessenov** [comment]
This is now ready, added more tests to recover Wasm coverage loss.

**Review by keith — APPROVED** [review]

**Comment by keith** [comment]
blah looks like you hit some more conflicts!

**Comment by moderation** [comment]
The ANTLR runtime is super out of date at `4.7.2` from 2018 - https://github.com/envoyproxy/envoy/blob/main/bazel/repository_locations.bzl#L1167-L1181=

Can this be upgraded to the latest release [`4.10.1`](https://github.com/antlr/antlr4/releases/tag/4.10.1)?

**Comment by kyessenov** [comment]
@moderation It is updated, it was just packaged as transitive dependency so the one listed in locations doesn't take effect (it wouldn't be possible because runtime API changed).

**Comment by moderation** [comment]
We might be talking about different ANTLR deps. This PR removed the `rules_antlr` "ANTLR Rules for Bazel" dependency (to transitive dependency). But the `antlr4_runtimes` depending still exists and is currently using `4.7.2`. I recall the latter being introduced with WASM /cc @PiotrSikora 
```python
        extensions = [
            "envoy.access_loggers.extension_filters.cel",
            "envoy.access_loggers.wasm",
            "envoy.bootstrap.wasm",
            "envoy.rate_limit_descriptors.expr",
            "envoy.filters.http.wasm",
            "envoy.filters.network.wasm",
            "envoy.stat_sinks.wasm",
        ],
```

**Comment by moderation** [comment]
Ignore this - just saw https://github.com/envoyproxy/envoy/pull/21608

**Comment by kyessenov** [comment]
@moderation Please see #21608

**Pull request merged**